zjit/src/codegen.rs

```diff
@@ -451,12 +451,10 @@ fn gen_getglobal(asm: &mut Assembler, id: ID) -> Opnd {
 }
 
 /// Set global variables
-fn gen_setglobal(asm: &mut Assembler, id: ID, val: Opnd) -> Opnd {
+fn gen_setglobal(asm: &mut Assembler, id: ID, val: Opnd) -> Option<()> {
     asm_comment!(asm, "call rb_gvar_set");
-    asm.ccall(
-        rb_gvar_set as *const u8,
-        vec![Opnd::UImm(id.0), val],
-    )
+    asm.ccall(rb_gvar_set as *const u8, vec![Opnd::UImm(id.0), val]);
+    Some(())
```

Comment:
Given that this does not have any failure paths, we can either return nothing or `()` here and handle (all of?) the type in the caller